src/pick.test.ts

```diff
@@ -42,14 +42,45 @@ test('read only', () => {
   pick(someObject, props); // TS2345 compilation error
 });
 
-test('type for curried form', () => {
-  const pickFoo = pick(['foo']);
+describe('curried form', () => {
+  test('it should pick props', () => {
+    const pickFoo = pick(['foo']);
+
+    expectTypeOf(true as unknown as ReturnType<typeof pickFoo>).toEqualTypeOf<{
+      foo: unknown;
+    }>();
 
-  expectTypeOf(true as any as ReturnType<typeof pickFoo>).toEqualTypeOf<
-    Record<'foo', any>
-  >();
+    const result = pickFoo({ foo: 1, bar: 'potato' });
 
-  const result = pickFoo({ foo: 1, bar: 'potato' });
+    expectTypeOf(result).toEqualTypeOf<{ foo: number }>();
+    expect(result).toEqual({ foo: 1 });
+  });
```

Comment:
Like in the omit case, we don't need to maintain the 'pickFoo' case so we don't need a test for it